### PR TITLE
added derivative of gamma

### PR DESCRIPTION
**Pull request by abhinavagarwal07**
TODO:
- [x] https://github.com/symengine/symengine/pull/752#discussion-diff-49275107

### CODE REVIEW

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -368,6 +367,12 @@ static RCP<const Basic> diff(const CLASS &self, \
             self.get_den())->diff(x));
     }
 
+    static RCP<const Basic> diff(const Gamma &self,
+            const RCP<const Symbol> &x) {
+        RCP<const Basic> gamma_arg = self.get_args()[0];
+        return mul(mul(gamma(gamma_arg), polygamma(zero, gamma_arg)), gamma_arg->diff(x));
```
Comment (isuruf) [inline symengine/derivative.cpp]: `gamma(gamma_arg)` can just be `self.rcp_from_this()`

Comment (abhinavagarwal07) [inline symengine/derivative.cpp]: @isuruf 
yes done

**Comment by isuruf** [comment]
Looks good. +1

**Comment by abhinavagarwal07** [comment]
@certik ping

**Comment by abhinavagarwal07** [comment]
@certik ping2

**Comment by rwst** [comment]
Fine with me, thanks.

**Pull request merged**